source/client/process_impl.h

```diff
@@ -68,6 +68,23 @@ class ProcessImpl : public Process, public Envoy::Logger::Loggable<Envoy::Logger
    */
   void shutdown() override;
 
+  /**
+   * Computes the SNI host based on the passed in uri(s), request headers, and protocol.
+   * Given a vector of Uris containing a single entry, its host fragment will be considered.
+   * Subsequently the passed request headers will be scanned for any host headers to override any
+   * Uri-derived sni host. If the passed in protocol is HTTP/2 or higher, next to host, :authority
+   * will be considered as well.
+   *
+   * @param parsed_uris The parsed target uris configured for the load test.
+   * @param request_headers Request headers to scan.
+   * @param protocol The anticipated protocol that will be used.
+   * @return std::string The sni-host derived from the configured load test target Uris and any
+   * host/authority request-headers found. Empty if no (unambiguous) sni host could be derived.
+   */
+  static std::string computeSniHost(const std::vector<UriPtr>& parsed_uris,
```

Comment:
Does this function in any way depend on the class we are attaching it to? If it doesn't can we place it in a pre-existing or a new relevant utility library or in an anonymous namespace within the same file? 

The latter would be preferred if nothing needs to call this from the outside, however we should keep the unit tests. Placing it outside of the class helps the reader understand that this doesn't depend on or modify the state of the class in any way.

Reply:
Sure, done in 84f83b7